### DISCUSSION TITLE
We don't need to Install the termux API

### DISCUSSION THREAD

**Discussion by khansaad1275** [khansaad1275/Termux-YTD#32] [General]
@cosmic3d 
Hey, why are we installing Termux API? 
The videos will appear in the gallary even if there is no Termux API on users termux 

below is the code, If we don't have any reason then we should remove this code. ( also don't show warning for no reson please)

# Install the termux API and inform the user about system gallery settings.
printf "${RED_B}WARNING!!! ${RESET}\n${YELLOW}By default, the videos you download won't appear in your system gallery, and therefore you won't be able to use them.\n"
printf "If you wish to see the video appear in your gallery, you'll have to install the ${MAGENTA}Termux:API app${RESET} via ${MAGENTA_B}${TERMUX_APK_RELEASE}${RESET}\n"

read -rp "Do you want to install the termux-api package? (yes/y/no/n) " RES

USER_ANS=$(echo "${RES^}" | cut -c 1-1 )

if [ $USER_ANS = "Y" ]; then
    printf "\n${CYAN}Installing termux-api package${RESET}\n"
    sleep 2
    pkg install termux-api
    if [ $? -eq 0 ]; then
  	  printf "${GREEN_B}termux-api package successfully installed${RESET}\n"
  	  printf "${YELLOW}Termux app was installed via ${MAGENTA_B}${TERMUX_APK_RELEASE}${RESET}\n"
      printf "${YELLOW}You will need to install the Termux:API app through that way too${RESET}\n"
	  else
  	  printf "${RED_B}An error occurred during termux-api installation${RESET}\n"
	  fi
    sleep 2
fi

**Comment by cosmic3d** [comment]
@khansaad1275 
The termux api package is needed because in a lot of android devices (like mine) there is a need for a media scan so the system gallery knows there are new files available, otherwise you won't be able to see/use them.

As you can see in the documentation, [termux-media-scan](https://wiki.termux.com/wiki/Termux-media-scan) is used for this purpose. 

![Screenshot_20240326_104336.jpg](https://github.com/khansaad1275/Termux-YTD/assets/68190267/f506f7d9-2125-44b7-ba44-d269edd5ed12)
